tests/tests_signin_user.py

```diff
@@ -0,0 +1,10 @@
+def test_with_authenticated_client(client, django_user_model):
+	username = "test"
+	password = "test"
+	user = django_user_model.objects.create_user(username=username, password=password)
+	client.force_login(user)
+	respons1 = client.get('/sign_in')
+	respons2 = client.get('/personal_area')
+	assert respons1.status_code == 301
+	assert respons2.status_code == 301
```

Comment:
Shouldn't you be getting a 200 for respons2.status_code?

Reply:
No, it is redirecting

Reply:
@BarBenNaim as per @simzacks  the second request should return 200, if you are already signed in, you need to check it

Reply:
I think because of that:
def sign_in_view(request):
    if request.method == 'POST':
        form = AuthenticationForm(data=request.POST)
        if form.is_valid():
            user = form.get_user()
            login(request, user)
            return redirect('personal_area')
    else:
        form = AuthenticationForm()
    return render(request, r"sign_in.html", {'form': form})


@login_required(login_url="/sign_in")
def personal_area(request):
    return render(request, r"personal_area.html")
@Yarboa @simzacks